File: src/services/api_service.py
import requests

API_URL = "https://jsonplaceholder.typicode.com"

def get_users():
    """Obtém a lista de usuários da API."""
    try:
        response = requests.get(f"{API_URL}/users")
        response.raise_for_status()
        return response.json()
    except requests.exceptions.RequestException as e:
        print(f"Erro ao fazer a requisição: {e}")
        return None
    
def get_user_details(user_id):
    """Obtém os detalhes de um usuário específico (nome e e-mail)."""
    try:
        response = requests.get(f"{API_URL}/users/{user_id}")
        response.raise_for_status()
        return response.json()
    except requests.exceptions.RequestException as e:
        print(f"Erro ao fazer a requisição: {e}")
        return None
    
def get_posts_by_user(user_id):
    """Obtém os posts de um usuário específico."""
    try:
        response = requests.get(f"{API_URL}/posts?userId={user_id}")
        response.raise_for_status()
        return response.json()
    except requests.exceptions.RequestException as e:
        print(f"Erro ao fazer a requisição: {e}")
        return None

File: frontend/src/api/apiService.js
const API_URL = "http://localhost:5000";

export const getUsers = async () => {
  try {
    const response = await fetch(`${API_URL}/users`);
    console.log("Response status:", response.status);
    if (!response.ok) {
      throw new Error("Erro ao obter usuários.");
    }
    const data = await response.json();
    console.log('Users data:', data);
    return data;
  } catch (error) {
    console.error("Error fetching users:", error);
    return null;
  }
};


export const getUserDetails = async (userId) => {
  try {
    const response = await fetch(`${API_URL}/users/${userId}`);
    if (!response.ok) {
      throw new Error("Erro ao obter detalhes do usuário.");
    }
    return await response.json();
  } catch (error) {
    console.error(error);
    return null;
  }
};


export const getUserPosts = async (userId) => {
  try {
    const response = await fetch(`${API_URL}/users/${userId}/posts`);
    if (!response.ok) {
      throw new Error("Erro ao obter posts do usuário.");
    }
    return await response.json();
  } catch (error) {
    console.error(error);
    return null;
  }
};